sources/packages/backend/libs/services/src/msfaa-number/msfaa-number-shared.service.ts

```diff
@@ -94,28 +94,73 @@ export class MSFAANumberSharedService {
         APPLICATION_INVALID_DATA_TO_CREATE_MSFAA_ERROR,
       );
     }
-    return this.internalCreateMSFAANumber(
+    return this.internalActivateMSFAANumber(
       application.student.id,
       application.id,
       application.currentAssessment.offering.offeringIntensity,
       auditUserId,
     );
   }
 
+  /**
+   *
+   * @param studentId student for which the MSFAA record will be reactivated.
+   * @param referenceApplicationId reference application id.
+   * @param offeringIntensity offering intensity.
+   * @param auditUserId user causing the change.
+   * @param msfaaNumber msfaaNumber to be reactivated.
+   * @param dateSigned date the msfaaNumber got signed.
+   * @param serviceProviderReceivedDate serviceProviderReceivedDate date.
+   * @returns reactivated MSFAA record.
+   */
+  async reactivateMSFAANumber(
+    studentId: number,
+    referenceApplicationId: number,
+    offeringIntensity: OfferingIntensity,
+    auditUserId: number,
+    msfaaNumber: Partial<MSFAANumber>,
```

Comment:
The only two properties from MSFAANumber used are `id` and `msfaaNumber`. If we consider that the `msfaaNumber` is used only for logging purposes (see https://github.com/bcgov/SIMS/pull/2010/files#r1238970760) maybe we just need the `id` here.

Reply:
Is it not the case to use the `Pick<MSFAA` here also?
If used in both places we can have a type defined also as below.
```ts
export type MSFAAActivationInfo = Pick<
  MSFAANumber,
  "id" | "msfaaNumber" | "dateSigned" | "serviceProviderReceivedDate"
>;
```
And then have the parameters of the method using it as below.
```ts
 async reactivateMSFAANumber(
    studentId: number,
    referenceApplicationId: number,
    offeringIntensity: OfferingIntensity,
    auditUserId: number,
    msfaaNumber: MSFAAActivationInfo,
    dateSigned: Date,
    serviceProviderReceivedDate: Date,
  ): Promise<MSFAANumber> {
```